test/specs/string-checker.js

```diff
@@ -357,6 +357,32 @@ describe('modules/string-checker', function() {
         });
     });
 
+    describe('throwing rules', function() {
+        beforeEach(function() {
+            checker = new StringChecker();
+            // register rule that throw
+            checker.registerRule({
+                configure : function() {},
+                getOptionName: function() { return 'thrower'; },
+                check : function() {
+                    throw Error('Here we are!');
+                }
+            });
+            checker.configure({
+                thrower: true
+            });
+        });
+
+        it('should be handled internally', function() {
+            var errs = checker.checkString('var a');
+            assert.equal(errs.getErrorCount(), 1);
+
+            var err = errs.getErrorList()[0];
+            assert.equal(err.rule, 'thrower');
+            assert.equal(err.message, 'Error: Here we are!');
```

Comment:
Fixing. I'm not sure that we should add it as an usual codestyle error. Moreover, I'm sure that we shouldn't use usual `errors.add` here. Probably only with `warn` level?